src/common/string/compare.ts

```diff
@@ -11,3 +11,21 @@ export const compare = (a: string, b: string) => {
 
 export const caseInsensitiveCompare = (a: string, b: string) =>
   compare(a.toLowerCase(), b.toLowerCase());
+
+export const fuzzyCompare = (a: string, b: string) => {
```

Comment:
I stuck with "a" and "b" convention here, but would it be better to use more descriptive vars here to make the recursion logic clearer?